app/views/common/os_selection/_pxe_loader.html.erb

```diff
@@ -0,0 +1,12 @@
+<%= fields_for item do |f| -%>
+  <% loaders = @operatingsystem.available_loaders(item.architecture); if loaders.any? -%>
+    <% selected_attr = f.object.new_record? && item.hostgroup.present? && item.hostgroup.pxe_loader.nil? ? { :selected => @operatingsystem.preferred_loader(item.architecture) } : {} -%>
```

Comment:
Ok pushing the change, also amending few fixes from above. The code is much more simple now. But I still see the error you described (not the error but suggested loader overrides old one). I will take a look tomorrow.